test/integration/control.js

```diff
@@ -0,0 +1,35 @@
+var path = require('path');
+var test = require('tap').test;
+var extract = require('../fixtures/extract');
+var VirtualMachine = require('../../src/index');
+
+var uri = path.resolve(__dirname, '../fixtures/control.sb2');
+var project = extract(uri);
+
+test('control project', function (t) {
+    var vm = new VirtualMachine();
+
+    // Evaluate playground data and exit
+    vm.on('playgroundData', function (e) {
+        var threads = JSON.parse(e.threads);
+        t.ok(threads.length > 0);
+        t.end();
+        process.nextTick(process.exit);
```

Comment:
Is this `process.exit` here because the top-level test function will have exited already? Does `tap` run each test in a separate process?

Instead of calling `process.exit` like this I think you can return a `Promise` from your top-level function, then resolve the promise instead of calling `t.end()` once the test is complete. I think that's the "official" way to deal with async tests, anyway. Some of the tests in `scratch-storage` work this way, though they're more complicated than this one would be.